.github/workflows/check.yml

```diff
@@ -0,0 +1,65 @@
+name: Combined Checks
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  combined-checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies for mypy
+      if: matrix.python-version != '3.8'
+      run: pip install mypy
+
+    - name: Configure mypy globally
+      if: matrix.python-version != '3.8'
+      run: echo -e "[mypy]\nignore_missing_imports = True" > mypy.ini
+
+    - name: Run mypy
+      if: matrix.python-version != '3.8'
```

Comment:
@zware 

Python 3.9 introduced enhancements to type hinting such as allowing built-in collection types (list, dict, set, tuple) to be used as generic types instead of using the typing module equivalents (List, Dict, Set, Tuple). As a result we would need to find a way to do different hinting for python 3.8 versus 3.9. 

Figured it would be easy to get type hinting out the door for >3.9 and if needed we could add to the roadmap adding in 3.8 support.

Reply:
we can and probably should use `from __future__ import annotations` that will leave them in their string form for all versions and you can use the newest forms as long as they parse correctly in the version of Python being run (so you can't use Python 3.12's [PEP-695](https://peps.python.org/pep-0695/) type syntax), but you can use built-in collection types since `list[Any]` is not a parse error as long as you use it only in the annotation context (you could not however use it in a runtime context like `cast(list[Any], my_list)`